rir/src/utils/configurations.h

```diff
@@ -16,7 +16,14 @@ struct Optimization {
 
     bool operator<(const Optimization& anotherOptimization) const {
         if (order == anotherOptimization.order) {
-            return anotherOptimization.translator->getName() == "cleanup";
+            if (anotherOptimization.translator->getName() == "cleanup") {
+                return true;
```

Comment:
I believe it is still wrong. For example now `(1, "cleanup") < (1, "cleanup")` :)

sorry, but the set really relies on on the order being an order. otherwise it will start overriding elements, those are really nasty bugs.

best thing here is to rely on somebody else being more clever. (ie. delegate to the stdlib). For example if you look at [tuple](http://en.cppreference.com/w/cpp/utility/tuple/operator_cmp) then you see that the generic implementation is actually kind of tricky:

```
(bool)(std::get<0>(lhs) < std::get<0>(rhs)) || (!(bool)(std::get<0>(rhs) < std::get<0>(lhs)) && lhstail < rhstail), where lhstail is lhs without its first element, and rhstail is rhs without its first element. For two empty tuples, 
```

so to use the tuple, try this:
```
auto T = [&](const Optimization& o) {
  return std::tuple<unsigned, unsigned, string>(o.order, o.getName() == "cleanup" ? 0 : 1, o.getName());
};
return T(*this) < T(other);
```

Reply:
So i guess just adding this condition should make it: